flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/AdaptiveScheduler.java

```diff
@@ -248,9 +250,24 @@ public AdaptiveScheduler(
                         declarativeSlotPool::reserveFreeSlot,
                         declarativeSlotPool::freeReservedSlot);
 
-        for (JobVertex vertex : jobGraph.getVertices()) {
-            if (vertex.getParallelism() == ExecutionConfig.PARALLELISM_DEFAULT) {
-                vertex.setParallelism(1);
+        if (configuration.get(JobManagerOptions.SCHEDULER_MODE)
+                == SchedulerExecutionMode.REACTIVE) {
+            LOG.info("Modifying job parallelism for running in reactive mode.");
+            for (JobVertex vertex : jobGraph.getVertices()) {
+                if (vertex.getMaxParallelism() == JobVertex.MAX_PARALLELISM_DEFAULT) {
+                    vertex.setParallelism(Transformation.UPPER_BOUND_MAX_PARALLELISM);
```

Comment:
Didn't we say in the FLIP that the reactive mode only works if no parallelism has been set?

Reply:
I'm not sure if we can detect whether the user has set the parallelism or not. At least the LocalStreamEnvironment and the TestStreamEnvironment (when used with MiniClusterWithClientResource) are setting a default parallelism in the ExecutionConfig, which will be set by the Api to JobGraph translation process. (In case of the LocalStreamEnvironment, it'll be the number of CPU cores, in TestStreamEnvironment, the number of slots)